src/Iact1dUnbinnedLkl.cc

```diff
@@ -889,64 +889,78 @@ Int_t Iact1dUnbinnedLkl::ReaddNdESignal(TString filename)
   return status;
 }
 
-Int_t Iact1dUnbinnedLkl::ReaddNdESignal(TString* filenames, Float_t* branchingRatios, Int_t numFiles)
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+// 
+// Read multiple dN_i/dE for signal from file in
+// the Segue Stereo input format produced by Jelena
+// and their corresponding branching ratios BR_i.
+// Then perform a linear combination
+// dN/dE = sum(BR_i * dN_i/dE).
+// Replacement of existing file is allowed
+// Return 0 in case of success
+//        1 if file is not found
+//
+Int_t Iact1dUnbinnedLkl::ReaddNdESignal(Int_t nFiles, TString* filenames, Float_t* branchingRatios)
 {
-  // open file and look for histo
-  TFile* dNdESignalFile[numFiles];
-  TH1F*  hProvdNdESignal[numFiles];
+  // Initialize the array of pointers and other variables
+  TFile* dNdESignalFile[nFiles];
+  TH1F*  hProvdNdESignal[nFiles];
   TH1F*  linCHdNdESignal;
 
   Int_t status      = 0;
   Int_t nBins       = 0;
   Double_t lowEdge  = 0.0;
   Double_t highEdge = 0.0;
-  for(Int_t file=0;file<numFiles;file++)
+  // open file and look for histograms
```

Comment:
In general, please a bit more of comments, e.g. here you should add "check that all histograms have same binning"